collections/index.js

```diff
@@ -20,46 +20,130 @@ import employees from './data.json';
+    // I thought of because if more than one value is empty per employee, this
+    // function only adds the employee to the array once.
+    if (employees[employeeNumberIndex].id === "" || employees[employeeNumberIndex].firstName === "" || employees[employeeNumberIndex].lastName === "" || employees[employeeNumberIndex].department === "" || employees[employeeNumberIndex].contactNumber === "" || employees[employeeNumberIndex].taxCode === "" || employees[employeeNumberIndex].salary === "") {
+      // this adds (pushes) the employees details to the missingInfoEmployees
+      // array we created earlier.
+      missingInfoEmployees.push(employees[employeeNumberIndex]);
+    }
+  }
+  // returns all employees that are missing data
+  return missingInfoEmployees;
+}
+
+// I thought of a 2nd way of doing this which is probably better but might
+// need some tidying up.
+
+export const getEmployeesWithMissingData = () => {
+  // sets an empty array ready to accept any values we add later:
+  var missingInfoEmployees = [];
+  // using a for loop to index through each of the employees on record
+  for (let employeeNumberIndex = 0; employeeNumberIndex < employees.length; employeeNumberIndex++) {
+    // creates an array of the employee details
+    let employeeDetails = Object.values(employees[employeeNumberIndex])
+    // checks each value in the new array for any empty strings
+    employeeDetails.find(function(item) {
+      if (item === "") {
+        // adds the employee to the missingInfoEmployees array
+        missingInfoEmployees.push(employees[employeeNumberIndex]);
+      }
+    })
+  }
+  // returns all employees that are missing data
+  return missingInfoEmployees;
 }
 
 /*
   I expect this functon to return me the employee list but with any sensitive
-  information like personal phone numbers or salary inforamtion stripped out
+  information like personal phone numbers or salary and tax inforamtion
+  stripped out
 */
 export const getNonSensitiveInformation = () => {
-
+  // using a for loop to index through each of the employees on record
+  for (let employeeNumberIndex = 0; employeeNumberIndex < employees.length; employeeNumberIndex++) {
+    // deletes the 3 sensitive key/value pairs from the employee records
+    delete employees[employeeNumberIndex].contactNumber;
+    delete employees[employeeNumberIndex].taxCode;
+    delete employees[employeeNumberIndex].salary;
+  }
+  // returns the ammended array with sensitive info removed. 
+  return employees;
 }
 
 /*
   I expect this function to return the total cost of all the employees salaries
 */
 export const calculateTotalSalary = () => {
-
+  var individualSalaries = [];
+  for (let employeeNumberIndex = 0; employeeNumberIndex < employees.length; employeeNumberIndex++) {
+    individualSalaries.push(employees[employeeNumberIndex].salary);
+  }
+  return individualSalaries.reduce(function(previous, current) {
+    return previous + current;
+  }, 0)
```

Comment:
Nice! 

We can actually simplify this a bit though 😏 

Rather than looping the array of employees to create an array of their salaries - why don't you call `.reduce()` directly on the employees?